### ISSUE TITLE
COSMIC applications very laggy upon resizing

### ISSUE THREAD

**Issue by BrodieRobertson** [pop-os/libcosmic#753]
Resizing COSMIC applications appears to be incredibly laggy, this does not occur with applications from any other toolkit.

https://github.com/user-attachments/assets/870aca99-114a-416d-801b-85a3d770ef2a

**Comment by Adam-Cosner** [comment]
This is probably because of iced's full redrawing on every update, which should be improved once iced 0.14 comes out and libcosmic switches to it

**Comment by lazytanuki** [comment]
> This is probably because of iced's full redrawing on every update, which should be improved once iced 0.14 comes out and libcosmic switches to it

I don't think it is the main cause, the issue wasn't present in the previous alphas.

**Comment by git-f0x** [comment]
It was present for me for quite a while, I think even before the first alpha.

**Comment by idoric** [comment]
> This is probably because of iced's full redrawing on every update, which should be improved once iced 0.14 comes out and libcosmic switches to it

@Adam-Cosner What iced’s improvement are you referring to?

**Comment by Adam-Cosner** [comment]
@idoric Iced 0.14 added Reactive Rendering, when iced 0.14 fully releases and COSMIC integrates that it'll reduce resource usage

**Comment by lazytanuki** [comment]
When resizing the window, widgets will need as much redrawing with reactive rendering as they do now, since this updates the widgets.
In addition, the number of displayed widgets doesn't seem to have an impact on this issue. Resizing an empty COSMIC app window lags as much as resizing COSMIC files for instance.
So I don't think reactive rendering will fix this.

**Comment by Adam-Cosner** [comment]
Aha! At last I've discovered the cause of the issue. This is an issue with iced's WGPU implementation

**Comment by Adam-Cosner** [comment]
This appears to have been fixed in iced 0.14, so we just need for that rebase to happen

**Comment by Adam-Cosner** [comment]
Correction, on my investigation this is due to most libcosmic apps using the iGPU, which doesn't handle resizing very well.

**Comment by lazytanuki** [comment]
The issue is also present on desktop PC.
Btw I can confirm the issue wasn't present in Alpha 3, at least for me.

**Comment by Adam-Cosner** [comment]
Latest commits for cosmic-comp should have fixed this

**Comment by Quackdoc** [comment]
I am testing on https://github.com/pop-os/cosmic-comp/commit/53d2cf01aa6578c550363c0da4cf1df941508424 for cosmic-comp and have the same issue with resizing still.

**Comment by Adam-Cosner** [comment]
@Quackdoc is the performance at least a bit better?

**Comment by Quackdoc** [comment]
going back and forth I can tell there is a difference, but there is still considerably lag.

**Comment by BrodieRobertson** [comment]
Just tested on my system and the resizing appears to be mostly functional now (not running at 3 FPS) but it does still feel kind of jittery.

**Comment by git-f0x** [comment]
Is it fixed now?

**Comment by Quackdoc** [comment]
for me at least it is on a 4k 60fps display with 150% scaling. Resizing is smooth.
